src/frontend/js/components/CourseAddToWishlist/index.tsx

```diff
@@ -0,0 +1,128 @@
+  LOG_ME = 'logMe',
+}
+const messages = defineMessages({
+  [CourseAddToWishlistMessageIds.LABEL_ADD]: {
+    id: 'components.CourseAddToWishlist.labelAdd',
+    description: 'label to proceed to add a course to user wishlist',
+    defaultMessage: 'Notify me',
+  },
+  [CourseAddToWishlistMessageIds.LABEL_REMOVE]: {
+    id: 'components.CourseAddToWishlist.labelRemove',
+    description: 'label to proceed to remove a course to user wishlist',
+    defaultMessage: 'Do not notify me',
+  },
+  [CourseAddToWishlistMessageIds.ERROR_ADD]: {
+    id: 'components.CourseAddToWishlist.errorAdd',
+    defaultMessage:
+      'An error occurred when adding this course to your wishlist. Please retry later.',
+    description: 'Error message shown when add to wishlist request failed.',
+  },
+  [CourseAddToWishlistMessageIds.ERROR_REMOVE]: {
+    id: 'components.CourseAddToWishlist.errorRemove',
+    defaultMessage:
+      'An error occurred when removing this course to your wishlist. Please retry later.',
+    description: 'Error message shown when remove to wishlist request failed.',
+  },
+  [CourseAddToWishlistMessageIds.LOG_ME]: {
+    id: 'components.CourseAddToWishlist.logMe',
+    defaultMessage: 'Log in to be notify',
+    description: 'label to proceed to login page before beeing adding a course to user wishlist.',
+  },
+});
+
+enum ComponentStates {
+  INITIALIZING = 'initializing',
+  LOADING = 'loading',
+  IDLE = 'idle',
+  ERROR = 'error',
+}
+
+export interface Props {
+  courseCode: Joanie.Course['code'];
+}
+
+const CourseAddToWishlist = ({ courseCode }: Props) => {
+  const [componentState, setComponentState] = useState<ComponentStates>(
+    ComponentStates.INITIALIZING,
+  );
+  const [isInWishlist, setIsInWishlist] = useState<boolean>(false);
+  const [errorMessageId, setErrorMessageId] = useState<CourseAddToWishlistMessageIds>();
+
+  const { user, login } = useSession();
+  const wishlistCourse = useUserWishlistCourses({ courseCode });
+
+  useEffect(() => {
+    if (componentState === ComponentStates.INITIALIZING) {
+      if (!wishlistCourse.states.fetching) {
+        setIsInWishlist(wishlistCourse.items.length > 0);
+        setComponentState(ComponentStates.IDLE);
+      }
+    }
+  }, [wishlistCourse.states.fetching, componentState]);
+
+  const removeFromWishlist = () =>
+    wishlistCourse.methods.delete(courseCode, {
+      onSuccess: () => setIsInWishlist(false),
+      onError: () => {
+        setComponentState(ComponentStates.ERROR);
+        setErrorMessageId(CourseAddToWishlistMessageIds.ERROR_ADD);
+      },
+    });
+  const addToWishlist = () =>
+    wishlistCourse.methods.create(courseCode, {
+      onSuccess: () => setIsInWishlist(true),
+      onError: () => {
+        setComponentState(ComponentStates.ERROR);
+        setErrorMessageId(CourseAddToWishlistMessageIds.ERROR_REMOVE);
+      },
+    });
+
+  if (componentState === ComponentStates.INITIALIZING) return <Spinner />;
+  return user === undefined ? (
```

Comment:
user can have three values : 

- `undefined`: app does not know yet if user is authenticated or not, a request is executed
- `null`: user is not logged in
- `User`: user is looged in

```suggestion
  return user ? (
```